### PR TITLE
Export function to extract and validate AK from server

### PR DESCRIPTION
**Pull request by yawangwang**
We need to export a function from server to extract/validate AK, and populate GCE instance info into machine state so that GAV can consume.

### CODE REVIEW

File: server/verify.go [hunk 1]
```diff
@@ -183,6 +152,45 @@ func VerifyAttestation(attestation *pb.Attestation, opts VerifyOpts) (*pb.Machin
 	return nil, fmt.Errorf("attestation does not contain a supported quote")
 }
 
+// ParseAndValidateAK parses and validate AK cert in the attestation, and populate GCE instance info.
```
Comment (alexmwu) [inline server/verify.go:155]: Ideally, we have composable funcs where we can verify the attestation key against the Attestation proto and VerifyOpts and then extract info from it. So I suggest a bit more refactoring:
1. call this `ValidateAK`
2. Have `ValidateAK` just call `validateAKPub` and `validateAKCert`
3. refactor `validateAKPub` and `validateAKCert` to only return error
4. refactor `validateAKCert` to not call `getInstanceInfoFromExtensions`
5. in `VerifyAttestation` call `ValidateAK` and then `getInstanceInfoFromExtensions` (or `GetGCEInstanceInfo` if cert != nil)

**Review by alexmwu — CHANGES_REQUESTED** [review]

File: server/verify_test.go [hunk 2]
```diff
@@ -651,56 +646,65 @@ func TestVerifyFailWithCertsAndPubkey(t *testing.T) {
...
+		},
+		{
+			name:     "failed with missing roots",
+			opts:     VerifyOpts{IntermediateCerts: GceEKIntermediates},
+			wantPass: false,
+		},
+		{
+			name:     "failed with missing intermediates",
+			opts:     VerifyOpts{TrustedRootCerts: GceEKRoots},
+			wantPass: false,
```
Comment (alexmwu) [inline server/verify_test.go:695]: Add a test case setting TrustedAKs in VerifyOpts but not Trusted/Intermediate certs

**Review by alexmwu — CHANGES_REQUESTED** [review]

File: server/verify.go (hunk 1 above)
Comment (yawangwang) [reply server/verify.go]: Done

File: server/verify_test.go (hunk 2 above)
Comment (yawangwang) [reply server/verify_test.go:723]: Done

**Review by alexmwu — APPROVED** [review]
Please give my changes a look @yawangwang and merge it in if you feel it looks okay.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -77,11 +77,12 @@ jobs:
         run: |
           GO_EXECUTABLE_PATH=$(which go)
           sudo $GO_EXECUTABLE_PATH test -v -run "TestFetchImageSignaturesDockerPublic" ./launcher
+          sudo $GO_EXECUTABLE_PATH test -v -run "TestHwAttestationPass" ./cmd
```
[the remark this answers is not in the document]
Comment (yawangwang) [reply .github/workflows/ci.yml]: See b/368161073, it returns `couldn't create report entry in configfs, no such device` error

**Comment by yawangwang** [comment]
> Please give my changes a look @yawangwang and merge it in if you feel it looks okay.

LGTM, now merging this PR

**Pull request merged**